src/services/PnsService.ts

```diff
@@ -0,0 +1,153 @@
+import { ethers } from 'ethers'
+import { SDK as PNSSDK } from 'pns-sdk'
+import { Chains } from 'pns-sdk/lib/constants'
+
+import { setInterceptor } from '../errors/ErrorInterceptor'
+import { AllDIDErrorCode, UnsupportedMethodError } from '../errors/AllDIDError'
+import { NamingService, RecordItem, RecordItemAddr } from './NamingService'
+import { makeRecordItem } from './UnsService'
+
+import type { Signer } from '@ethersproject/abstract-signer'
```

Comment:
The `@ethersproject/abstract-signer` module does not seem to be declared in `package.json`.